### PR TITLE
Update TextStyle and StrutStyle height docs

### PR DESCRIPTION
**Pull request by GaryQian**
This docs change reflects the changes to `TextStyle.height` and `StrutStyle.height` we are proposing to make.

See https://github.com/flutter/flutter/issues/29479 and https://github.com/flutter/engine/pull/9041

`height` default behavior is unchanged, however, when provided, the value now produces line height that is truly `height * fontSize` tall. Before, we were multiplying the font-provided ascent and descent, which did not necessarily add up to the font size.

This should not be landed until the engine-side PR lands and rolls.

### CODE REVIEW

**Review by Piinks — COMMENTED** [review]
Is there a link to the breaking change announcement?

**Comment by GaryQian** [comment]
Not yet, haven't written it yet. Writing it now!

**Comment by GaryQian** [comment]
Breaking change proposal sent to flutter-announce. The migration for this one may be harder than it might initially seem, so if anyone has any problems with doing so please let me know! I'll see if I can write a more comprehensive guide to doing so.

File: packages/flutter/lib/src/painting/strut_style.dart
Comment (goderbauer) [inline packages/flutter/lib/src/painting/strut_style.dart]: nit: null shouldn't be enclosed in `

File: packages/flutter/lib/src/painting/text_style.dart
Comment (goderbauer) [inline packages/flutter/lib/src/painting/text_style.dart]: null should not be enclosed in `

File: packages/flutter/lib/src/painting/strut_style.dart
Comment (dnfield) [inline packages/flutter/lib/src/painting/strut_style.dart]: nit: can we make these lines <80 characters?

File: packages/flutter/lib/src/painting/strut_style.dart
Comment (dnfield) [inline packages/flutter/lib/src/painting/strut_style.dart]: Can we have a diagram of this?

File: packages/flutter/lib/src/painting/strut_style.dart
Comment (goderbauer) [inline packages/flutter/lib/src/painting/strut_style.dart]: If height is a property (which I think it is), this should be [height]. Same for fontSize.

File: packages/flutter/lib/src/painting/strut_style.dart
Comment (dnfield) [reply packages/flutter/lib/src/painting/strut_style.dart]: (Specifically: a diagram of a character or string that shows the ascent, descent, leading)

File: packages/flutter/lib/src/painting/strut_style.dart
Comment (goderbauer) [inline packages/flutter/lib/src/painting/strut_style.dart]: "ascent + decent" enclosed in ` like the multiplication below or write it as prose?

File: packages/flutter/lib/src/painting/strut_style.dart
Comment (goderbauer) [inline packages/flutter/lib/src/painting/strut_style.dart]: missing a `.` before The?

File: packages/flutter/lib/src/painting/text_style.dart
Comment (goderbauer) [inline packages/flutter/lib/src/painting/text_style.dart]: [height] and [fontSize]?

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/painting/strut_style.dart
Comment (dnfield) [inline packages/flutter/lib/src/painting/strut_style.dart]: Am I missing a diagram somewhere?  Or even just a wikipedia link that has one?

Comment (GaryQian) [reply packages/flutter/lib/src/painting/strut_style.dart]: Making one :)

Comment (GaryQian) [reply packages/flutter/lib/src/painting/strut_style.dart]: Diagram is at https://github.com/flutter/assets-for-api-docs/pull/86

Comment (GaryQian) [reply packages/flutter/lib/src/painting/strut_style.dart]: Will update the docs to reference it once it lands.

Comment (GaryQian) [reply packages/flutter/lib/src/painting/strut_style.dart]: https://github.com/flutter/assets-for-api-docs/pull/87

File: packages/flutter/lib/src/painting/strut_style.dart
Comment (GaryQian) [reply packages/flutter/lib/src/painting/strut_style.dart]: In order to better explain strut, I think i'll make a separate set of diagrams in a dedicated PR general to all of strut since we need to use the concept of ascent and descent to fully/properly explain its behavior.

**Review by Piinks — APPROVED** [review]
LGTM! Awesome diagram!

**Comment by GaryQian** [comment]
Update: Holding off on committing this until the tree is in a position to take riskier changes. Due to the very breaking nature of this change, the delay will allow for a much more flexible and longer period of time to work with breakages that this causes.

**Comment by chunhtai** [comment]
why is this a breaking change? i only saw comment updates in the diff

**Comment by GaryQian** [comment]
The implementation is changing in the engine-side patch (flutter/engine#9041)

**Comment by GaryQian** [comment]
new diagram: https://github.com/flutter/assets-for-api-docs/pull/90

**Comment by GaryQian** [comment]
Engine patch is part of https://github.com/flutter/flutter/pull/35814

**Pull request merged**
